### PR TITLE
system-variables: add tiflash_hashagg_preaggregation_mode

### PR DESCRIPTION
**Pull request by guo-shaoge**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -5189,6 +5189,18 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
...
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：是
+- 默认值：`auto`
+- 可选值：`auto`、`force_streaming`、`force_preagg`
+- 该变量用于控制下推到 TiFlash 的两阶段或者三阶段 HashAgg 的第一阶段采用哪种预聚合策略：
+    - `force_preagg`: 表示 TiFlash 会在第一阶段的 HashAgg 进行强制预聚合操作
+    - `force_streaming`: 表示 TiFlash 会直接将数据发送给下一阶段的 HashAgg ，不进行预聚合操作
+    - `auto`: 表示 TiFlash 会自动根据所看到的 workload 的聚合度进行判断是否进行预聚合
```
Comment (Oreoxmt) [inline system-variables.md:5202]:
```suggestion
- 可选值：`auto`、`force_preagg`、`force_streaming`
- 该变量用于控制下推到 TiFlash 的两阶段或三阶段 HashAgg 的第一阶段采用哪种预聚合策略：
    - `auto`：TiFlash 根据当前工作负载的聚合度自动选择是否进行预聚合操作
    - `force_preagg`：TiFlash 在第一阶段的 HashAgg 中强制进行预聚合操作
    - `force_streaming`：TiFlash 直接将数据发送到下一阶段的 HashAgg，不进行预聚合操作
```

File: system-variables.md [hunk 2]
```diff
@@ -5189,6 +5189,18 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
 > - 如果 TiDB 节点未设置[区域属性](/schedule-replicas-by-topology-labels.md#设置-tidb-的-labels可选)，并且 TiFlash 副本选择策略不是 `all_replicas` 时，TiFlash 引擎将忽略 TiFlash 副本选择策略，使用所有 TiFlash 副本进行 TiFlash 查询，并且返回警告 `The variable tiflash_replica_read is ignored`。
 > - 如果 TiFlash 节点未设置[区域属性](/schedule-replicas-by-topology-labels.md#设置-tikv-和-tiflash-的-labels)，则将其视为不属于任何区域的节点。
 
+### `tiflash_hashagg_preaggregation_mode` <span class="version-mark">从 v8.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：是
+- 默认值：`auto`
```
Comment (Oreoxmt) [inline system-variables.md:5197]:
```suggestion
- 类型：枚举型
- 默认值：`auto`
```

Comment (joey-yez) [inline system-variables.md:5197]: according to the spec, the default value should be  force_preagg, which means the same behavior as the previous version.

File: system-variables.md [hunk 3]
```diff
@@ -5189,6 +5189,19 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
...
+### `tiflash_hashagg_preaggregation_mode` <span class="version-mark">从 v8.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：是
+- 类型：字符串
+- 默认值：`force_preagg`
+- 可选值：`auto`、`force_streaming`、`force_preagg`
+- 该变量用于控制下推到 TiFlash 的两阶段或者三阶段 HashAgg 的第一阶段采用哪种预聚合策略：
+    - `force_preagg`: 表示 TiFlash 会在第一阶段的 HashAgg 进行强制预聚合操作
```
Comment (joey-yez) [inline system-variables.md:5201]:
```suggestion
    - `force_preagg`: 表示 TiFlash 会在第一阶段的 HashAgg 进行强制预聚合操作。引入该变量前，TiDB 第一阶段 HashAgg 使用该策略。
```

Comment (guo-shaoge) [reply system-variables.md]: done

File: system-variables.md (hunk 1 above)
Comment (guo-shaoge) [reply system-variables.md]: default value changed to `force_preagg`

File: system-variables.md [hunk 4]
```diff
@@ -5189,6 +5189,19 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
...
+### `tiflash_hashagg_preaggregation_mode` <span class="version-mark">从 v8.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：是
+- 类型：枚举型
+- 默认值：`force_preagg`
+- 可选值：`force_preagg`、`force_streaming`、`auto`
+- 该变量用于控制下推到 TiFlash 的两阶段或者三阶段 HashAgg 的第一阶段采用哪种预聚合策略：
+    - `force_preagg`: 表示 TiFlash 会在第一阶段的 HashAgg 进行强制预聚合操作。引入改变量前，TiFlash 的行为等效于该选项
```
Comment (joey-yez) [inline system-variables.md:5201]:
```suggestion
    - `force_preagg`: 表示 TiFlash 会在第一阶段的 HashAgg 进行强制预聚合操作。引入该变量前，TiFlash 的行为等效于该选项
```

**Review by joey-yez — APPROVED** [review]

File: system-variables.md [hunk 5]
```diff
@@ -5189,6 +5189,19 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
...
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：是
+- 类型：枚举型
+- 默认值：`force_preagg`
+- 可选值：`force_preagg`、`force_streaming`、`auto`
+- 该变量用于控制下推到 TiFlash 的两阶段或者三阶段 HashAgg 的第一阶段采用哪种预聚合策略：
+    - `force_preagg`: 表示 TiFlash 会在第一阶段的 HashAgg 进行强制预聚合操作。引入该变量前，TiFlash 的行为等效于该选项
+    - `force_streaming`: 表示 TiFlash 会直接将数据发送给下一阶段的 HashAgg ，不进行预聚合操作
+    - `auto`: 表示 TiFlash 会自动根据所看到的 workload 的聚合度进行判断是否进行预聚合
```
Comment (Oreoxmt) [inline system-variables.md:5203]:
```suggestion
- 该变量用于控制下推到 TiFlash 的两阶段或三阶段 HashAgg 的第一阶段采用哪种预聚合策略：
    - `force_preagg`：TiFlash 在第一阶段的 HashAgg 中强制进行预聚合操作，与 v8.3.0 之前版本的行为一致
    - `force_streaming`：TiFlash 直接将数据发送到下一阶段的 HashAgg，不进行预聚合操作
    - `auto`：TiFlash 根据当前工作负载的聚合度自动选择是否进行预聚合操作
```

**Review by Oreoxmt — APPROVED** [review]

File: system-variables.md [hunk 6]
```diff
@@ -5189,6 +5189,19 @@ Query OK, 0 rows affected, 1 warning (0.00 sec)
...
 
+### `tiflash_hashagg_preaggregation_mode` <span class="version-mark">从 v8.3.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：是
+- 类型：枚举型
+- 默认值：`force_preagg`
+- 可选值：`force_preagg`、`force_streaming`、`auto`
+- 该变量用于控制下推到 TiFlash 的两阶段或三阶段 HashAgg 的第一阶段采用哪种预聚合策略：
```
Comment (qiancai) [inline system-variables.md:5200]:
```suggestion
- 该变量用于控制 TiFlash 在执行两阶段或三阶段 HashAgg 时，第一阶段采用哪种预聚合策略：
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**